google/cloud/security/common/gcp_type/firewall_rule.py

```diff
@@ -205,14 +213,16 @@ def is_equivalent(self, other):
 class FirewallAction(object):
     """An association of allowed or denied ports and protocols."""
 
-    def __init__(self, firewall_rule_allowed=None, firewall_rule_denied=None):
+    def __init__(self, firewall_rule_allowed=None, firewall_rule_denied=None,
+                 firewall_rule_action='allow'):
         """Initialize.
 
         Args:
           firewall_rule_allowed (list): A list of dictionaries of allowed ports
             and protocols.
           firewall_rule_denied (list): A list of dictionaries of denied ports
             and protocols.
+          firewall_rule_action (str): The action, either allow or deny.
```

Comment:
Now that you have the action, does it make sense to just include firewall_rules and firewall_rule_action, and not separate out allow and deny rules into distinct arguments? It seems odd to currently be able to call FirewallAction(firewall_rule_denied=denied_rules) and have it be valid, even though firewall_rule_action is still defaulting to 'allow'.